crates/autopilot/src/solvable_orders.rs

```diff
@@ -283,6 +291,33 @@ impl SolvableOrdersCache {
     }
 }
 
+/// Applies fee objective scaling and discard all orders where an overflow
+/// occurred.
+fn apply_fee_objective_scaling_factor(
+    mut orders: Vec<Order>,
+    scaling_factor: &BigRational,
+) -> Vec<Order> {
+    orders.retain_mut(|order| {
+        if order.metadata.class == OrderClass::Liquidity {
+            // A liquidity order should only be used if it can increase a settlement's total
+            // surplus more than it costs to execute it.
+            // That's why we don't want to promote it artificially by scaling its fee.
+            return true;
+        }
+        let fee = u256_to_big_rational(&order.metadata.solver_fee);
+        let scaled_fee = fee * scaling_factor;
+        let scaled_fee = match big_rational_to_u256(&scaled_fee) {
+            Ok(fee) => fee,
+            Err(_) => return false,
```

Comment:
This should never happen in practice (so not requesting change here), but semantically, I would expect this to be a saturating operation and not a checked one (i.e. overflowing means `fee == U256::MAX` and not order doesn't get considered).

That being said, since we don't _expect_ this to happen, I think it makes slightly more sense to filter the order out, so we we a log and increment a metric (as it would indicate something terribly wrong with our current assumptions).